pkg/filter/driftignore.go

```diff
@@ -51,18 +52,19 @@ func (r *DriftIgnore) readIgnoreFile() error {
 		}
 		line = strings.ReplaceAll(line, "/", separator)
 
-		lines = append(lines, gitignore.ParsePattern(line, nil))
+		patterns = append(patterns, gitignore.ParsePattern(line, nil))
 		if !strings.HasSuffix(line, "*") {
 			line := fmt.Sprintf("%s.*", line)
-			lines = append(lines, gitignore.ParsePattern(line, nil))
+			patterns = append(patterns, gitignore.ParsePattern(line, nil))
 		}
+		r.lines = append(r.lines, line)
```

Comment:
Do we really need to store all the lines in memory ? Maybe a counter can be enough here like a `lineCount` or `rulesCount` as you named in in tests ?